### PR TITLE
My watchlist

### PR DESCRIPTION
**Pull request by Quentin-Macquart**

### CODE REVIEW

**Review by BenZen59 — DISMISSED** [review]

File: src/components/FicheFilm/FicheFilm.jsx [hunk 1]
```diff
@@ -46,8 +48,11 @@ export default function FicheFilm() {
         `https://api.themoviedb.org/3/movie/${id}/credits?api_key=599ded6f0fc3bcaee1882e83ae0d438a`
       )
       .then(({ data }) => {
-        setDirectors(data.crew[5]);
+        setDirector(data.crew[5]);
         setActors(data.cast);
+      })
+      .catch(() => {
+        console.error('Erreur API');
```
Comment (LoicBrassart) [inline src/components/FicheFilm/FicheFilm.jsx]: No `console.*`

File: src/components/FicheFilm/FicheFilm.jsx [hunk 2]
```diff
@@ -57,10 +62,36 @@ export default function FicheFilm() {
         `https://api.themoviedb.org/3/movie/${id}/videos?api_key=599ded6f0fc3bcaee1882e83ae0d438a`
       )
       .then(({ data }) => {
-        setTrailers(data.results[0]);
+        setTrailer(data.results[0]);
+      })
+      .catch(() => {
+        console.error('Erreur API');
```
Comment (LoicBrassart) [inline src/components/FicheFilm/FicheFilm.jsx]: Nope

File: src/components/FicheFilm/FicheFilm.jsx [hunk 3]
```diff
@@ -74,17 +105,37 @@ export default function FicheFilm() {
...
+  let UsersScorePictures;
+  if (voteAverage === 10) {
+    UsersScorePictures = star5;
+  } else if (voteAverage < 10 && voteAverage >= 8) {
+    UsersScorePictures = star4;
+  } else if (voteAverage < 8 && voteAverage >= 6) {
+    UsersScorePictures = star3;
+  } else if (voteAverage < 6 && voteAverage >= 4) {
+    UsersScorePictures = star2;
+  } else if (voteAverage < 4) {
```
Comment (LoicBrassart) [inline src/components/FicheFilm/FicheFilm.jsx]: Simplify these conditions

File: src/components/FicheFilm/FicheFilm.jsx [hunk 4]
```diff
@@ -74,17 +105,37 @@ export default function FicheFilm() {
...
+    UsersScorePictures = star1;
+  }
+
+  const hour = (runtime - (runtime % 60)) / 60;
+  const min = (runtime / 60 - hour) * 60;
```
Comment (LoicBrassart) [inline src/components/FicheFilm/FicheFilm.jsx]: `const min = runtime % 60;` (Yeah, this `%` operator is *really* useful :p )

File: src/components/FicheFilm/FicheFilm.jsx [hunk 5]
```diff
@@ -74,17 +105,37 @@ export default function FicheFilm() {
...
 
   return (
+
```
Comment (LoicBrassart) [inline src/components/FicheFilm/FicheFilm.jsx]: Useless empty line

File: src/components/FicheFilm/FicheFilm.jsx [hunk 6]
```diff
@@ -74,17 +105,37 @@ export default function FicheFilm() {
...
     <div className="movieMainBloc">
       <div className="titleVote">
         <p>{titles}</p>
         <p className="vote">
           <img className="starScore" src={UsersScorePictures} alt="starScore" />
           <br />
           <div className="numberVote">{numberVotes} votes</div>
+
```
Comment (LoicBrassart) [inline src/components/FicheFilm/FicheFilm.jsx]: Useless empty line

File: src/components/FicheFilm/FicheFilm.jsx [hunk 7]
```diff
@@ -118,19 +169,30 @@ export default function FicheFilm() {
...
+        ) : (
+          <button type="button" onClick={addToWatch} className="buttonAdd">
+            <span className="buttonText+or-">Add to Watch!</span>
+          </button>
+        )}
+
+        <a href={`https://www.youtube.com/embed/${trailer.key}`}>
+          <div className="buttonTrailer" alt="Trailer">
+
+
```
Comment (LoicBrassart) [inline src/components/FicheFilm/FicheFilm.jsx]: Useless empty lines + Prettier wouldn't have let this pass, what happened ?

File: src/components/FicheFilm/FicheFilm.jsx [hunk 8]
```diff
@@ -118,19 +169,30 @@ export default function FicheFilm() {
...
             <div className="playTriangle" />
           </div>
         </a>
       </div>
+
```
Comment (LoicBrassart) [inline src/components/FicheFilm/FicheFilm.jsx]: Nope

File: src/components/watchlist/watchlist.css [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  width: 50vw;
+  height: 400px;
+  overflow: hidden;
+  filter: blur(0) opacity(100%) brightness(100%);
+  object-fit: cover;
+  object-position: center;
+}
+
+.watchImg:hover {
+  z-index: -40;
```
Comment (LoicBrassart) [inline src/components/watchlist/watchlist.css]: Grmpf... Why ?

File: src/components/watchlist/watchlist.css [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+  opacity: 0.5;
+}
+
+.movieTitle {
+  z-index: -1;
```
Comment (LoicBrassart) [inline src/components/watchlist/watchlist.css]: ...

**Review by LoicBrassart — DISMISSED** [review]
Some Evils to fix

**Review by BenZen59 — DISMISSED** [review]

File: src/components/watchlist/watchlist.css (hunk 9 above)
Comment (Quentin-Macquart) [reply src/components/watchlist/watchlist.css]: I added corrections!

File: src/components/watchlist/watchlist.css (hunk 10 above)
Comment (Quentin-Macquart) [reply src/components/watchlist/watchlist.css]: I added corrections!

File: src/components/FicheFilm/FicheFilm.jsx (hunk 8 above)
Comment (Quentin-Macquart) [reply src/components/FicheFilm/FicheFilm.jsx]: I added corrections!

File: src/components/FicheFilm/FicheFilm.jsx (hunk 7 above)
Comment (Quentin-Macquart) [reply src/components/FicheFilm/FicheFilm.jsx]: I added corrections!

File: src/components/FicheFilm/FicheFilm.jsx (hunk 6 above)
Comment (Quentin-Macquart) [reply src/components/FicheFilm/FicheFilm.jsx]: I added corrections!

File: src/components/FicheFilm/FicheFilm.jsx (hunk 5 above)
Comment (Quentin-Macquart) [reply src/components/FicheFilm/FicheFilm.jsx]: I added corrections!

File: src/components/FicheFilm/FicheFilm.jsx (hunk 4 above)
Comment (Quentin-Macquart) [reply src/components/FicheFilm/FicheFilm.jsx]: I added corrections!

File: src/components/FicheFilm/FicheFilm.jsx (hunk 3 above)
Comment (Quentin-Macquart) [reply src/components/FicheFilm/FicheFilm.jsx]: I added corrections!

File: src/components/FicheFilm/FicheFilm.jsx (hunk 2 above)
Comment (Quentin-Macquart) [reply src/components/FicheFilm/FicheFilm.jsx]: I added corrections!

File: src/components/FicheFilm/FicheFilm.jsx (hunk 1 above)
Comment (Quentin-Macquart) [reply src/components/FicheFilm/FicheFilm.jsx]: I added corrections!

**Review by BenZen59 — APPROVED** [review]

**Review by Hacene010 — APPROVED** [review]

**Pull request merged**